tests/git-scripts/pre-push.sh

```diff
@@ -36,14 +36,13 @@ run_test()
 {
     TEST=$1
     echo "running '$TEST'"
-    if ! `$TEST > /dev/null 2>&1`; then
+    if ! tests/scripts/all.sh "$TEST" >/dev/null 2>&1; then
         echo "test '$TEST' failed"
         return 1
     fi
 }
 
-run_test ./tests/scripts/check-doxy-blocks.pl
-run_test ./tests/scripts/check-names.sh
-run_test ./tests/scripts/check-generated-files.sh
-run_test ./tests/scripts/check-files.py
-run_test ./tests/scripts/doxygen.sh
+# could direclty invoke "all.sh 'check_*'" but we want shorter output
+for check in $(tests/scripts/all.sh --list-components | grep '^check_'); do
```

Comment:
So how about `tests/scripts/all.sh -k 'check_*' | ! grep FAILED`?

Reply:
Maybe I'm overthinking this, but what if `all.sh` fails without writing `FAILED` on `stdout`? For example an unexpected error is caught by `set -e` in `all.sh`. In that case the `! grep` would succeed and the non-zero exit status from `all.sh` would be ignore.

Reply:
Good point. `{ tests/scripts/all.sh -k 'check_*' || echo "FAILED: unexpected failure in all.sh"; } | ! grep FAILED`

Reply:
Sounds good! I'll update the PR with that.

Reply:
Marking as resolved as the entire question is avoided by the new approach of with `all.sh -q`.